### PR TITLE
ath79: add support for I-O DATA WN-AC1600DGR2

### PR DESCRIPTION
**Pull request by musashino205**
I-O DATA WN-AC1600DGR2 is a 2.4/5 GHz band 11ac router, based on
Qualcomm Atheros QCA9557.

Specification:

- Qualcomm Atheros QCA9557
- 128 MB of RAM
- 16 MB of Flash
- 2.4/5 GHz wifi
  - 2.4 GHz: 2T2R (SoC internal)
  - 5 GHz: 3T3R (QCA9880)
- 5x 10/100/1000 Mbps Ethernet
- 6x LEDs, 6x keys (4x buttons, 1x slide switch)
- UART header on PCB
  - Vcc, GND, TX, RX from ethernet port side
  - 115200n8

Flash instruction using factory image:

1. Connect the computer to the LAN port of WN-AC1600DGR2
2. Connect power cable to WN-AC1600DGR2 and turn on it
3. Access to "http://192.168.0.1/" and open firmware update page
("ファームウェア")
4. Select the OpenWrt factory image and click update ("更新") button
5. Wait ~150 seconds to complete flashing

Signed-off-by: INAGAKI Hiroshi <musashino.open@gmail.com>

- Note: MAC address is stored as text only in u-boot-env.

### CODE REVIEW

File: target/linux/ath79/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac [hunk 1]
```diff
@@ -13,6 +13,9 @@ PHYNBR=${DEVPATH##*/phy}
 board=$(board_name)
 
 case "$board" in
+	iodata,wn-ac1600dgr2)
```
Comment (mkresin) [inline target/linux/ath79/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac]: please patch the mac address in the ath10k firmware instead.

**Review by mkresin — CHANGES_REQUESTED** [review]

File: target/linux/ath79/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac (hunk 1 above)
Comment (musashino205) [reply target/linux/ath79/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac]: Is it also for SoC internal wmac?

Comment (mkresin) [reply target/linux/ath79/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac]: Wait, I got confused. For which wireless you need to set the mac address? Isn't it the pcie ath10k wireless? Or does the SoC wifi has a wrong mac address as well and you don't know yet how to fix it?

Would you please paste the mac addresses of the stock firmware and provide a dump of the u-boot-env and art partition. Might be better if I have a look on it on my own before asking more questions and giving bad advices. Feel free to send it to me via mail.

Comment (musashino205) [reply target/linux/ath79/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac]: Oh, I'm sorry I confused you...
WN-AC1600DGR2 supported 2.4G (internal) and 5G (pcie) wifi, but mac addresses (ethaddr, wanaddr) is stored only in u-boot-env as far as I have investigated. So it is necessary to read from u-boot-env and set it for 2.4/5GHz wifi.

- MAC addresses
  - LAN: 34:76:C5:35:11:30
  - WAN: 34:76:C5:35:11:32
  - 2.4G: 34:76:C5:35:11:30
  - 5G: 34:76:C5:35:11:31

I sent an email with dump files attached.

Comment (mkresin) [reply target/linux/ath79/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac]: > Oh, I'm sorry I confused you...

No I failed to see that 10_fix_wifi_mac fixes the mac address of all wireless phys and that your board has a wmac as well. More will follow via e-mail.

**Comment by musashino205** [comment]
I updated this PR.

**Comment by mkresin** [comment]
Due to merge conflicts, I had to rework parts of your PR. Would you please build and runtime test an image build from [my staging tree](https://git.openwrt.org/?p=openwrt/staging/mkresin.git;a=summary) to ensure I didn't broke anything.

**Comment by musashino205** [comment]
OK, I'll do it.

**Comment by musashino205** [comment]
Done, everything worked fine without any problems ;)

**Comment by mkresin** [comment]
Perfect! Closing the PR now and will push my staging tree later the day.

**Pull request closed without merging**